ui/src/components/chat/compose/messages/effects.rs

```diff
@@ -0,0 +1,90 @@
+use crate::utils;
+use common::state::State;
+use dioxus_core::Scoped;
+use dioxus_hooks::{to_owned, use_effect, UseRef, UseSharedState};
+use uuid::Uuid;
+
+use crate::components::chat::compose::messages::scripts::{
+    SCROLL_BOTTOM, SCROLL_TO, SCROLL_UNREAD, SETUP_CONTEXT_PARENT,
+};
+
+use super::{get_messagesProps, NewelyFetchedMessages};
+
+pub fn update_chat_messages<'a>(
+    cx: &'a Scoped<'a, get_messagesProps>,
+    state: &UseSharedState<State>,
+    newely_fetched_messages: &UseRef<Option<NewelyFetchedMessages>>,
+) {
+    use_effect(cx, (), |_| {
+        to_owned![state, newely_fetched_messages];
+        async move {
+            if let Some(NewelyFetchedMessages {
+                conversation_id,
+                messages,
+                has_more,
+            }) = newely_fetched_messages.write_silent().take()
+            {
+                state
+                    .write()
+                    .update_chat_messages(conversation_id, messages);
+                if !has_more {
+                    log::debug!("finished loading chat: {conversation_id}");
+                    state.write().finished_loading_chat(conversation_id);
+                }
+            }
+        }
+    });
+}
+
+pub fn check_message_scroll<'a>(
+    cx: &'a Scoped<'a, get_messagesProps>,
+    scroll_to: &Option<Uuid>,
+    state: &UseSharedState<State>,
+    eval: &utils::EvalProvider,
+    currently_active: &Option<Uuid>,
+) {
+    use_effect(cx, scroll_to, |_| {
+        to_owned![state, eval, currently_active];
+        async move {
+            let currently_active = match currently_active {
+                Some(r) => r,
+                None => return,
+            };
+            if let Some(uuid) = state.write_silent().check_message_scroll(&currently_active) {
```

Comment:
Here, we need to use read(), right? 
```suggestion
            if let Some(uuid) = state.read().check_message_scroll(&currently_active) {
```

Reply:
> Here, we need to use read(), right?

If I recall, `check_message_scroll` is a mutable function so calling `read` would cause an error

Reply:
Hum... so it is changing a value inside state, and return other value to use in conditional, right?

Reply:
Darius is correct. `read()` is inappropriate here.